pandas/core/internals/blocks.py

```diff
@@ -3004,9 +3003,8 @@ def _extend_blocks(result, blocks=None):
                 blocks.extend(r)
             else:
                 blocks.append(r)
-    elif isinstance(result, BlockManager):
-        blocks.extend(result.blocks)
     else:
+        assert isinstance(result, Block), type(result)
```

Comment:
do we need a run-time check? can result be typed as Union[Block, List[Union[Block, List[Block]]]? Might help to type blocks and return type while touching this.

Reply:
thats a pretty ugly nested union; ill see if i can get rid of the double-list in the next pass

Reply:
@simonjayhawkins is this a deal-breaker?  id like to clear the queue